mobile/cmp/popover/Popover.js

```diff
@@ -0,0 +1,196 @@
+
+        // Respect controlled mode prop
+        if (!isNil(isOpen)) {
+            impl.setControlledMode(isOpen, onInteraction);
+        }
+
+        return div({
+            className,
+            items: [
+                div({
+                    ref: impl.targetRef,
+                    className: 'xh-popover__target-wrapper',
+                    item: target,
+                    onClick: () => impl.toggleOpen()
+                }),
+                div({
+                    ref: impl.contentRef,
+                    omit: !impl.isOpen,
+                    style: popper?.styles?.popper,
+                    className: 'xh-popover__content-wrapper',
+                    items: content
+                }),
+                div({
+                    omit: !impl.isOpen,
+                    className: classNames(
+                        'xh-popover__content-overlay',
+                        withMask ? 'xh-popover__content-overlay--mask' : null
+                    ),
+                    onClick: () => impl.setIsOpen(false)
+                })
+            ]
+        });
+    }
+});
+
+Popover.propTypes = {
+    /** Component to display inside the popover */
+    content: PT.element,
+
+    /** The target to which the popover content is attached */
+    target: PT.element,
+
+    /** Whether the popover is visible. Passing this prop puts the popover in controlled mode */
+    isOpen: PT.bool,
+
+    /**
+     * Callback invoked in controlled mode when the popover open state _would_ change due to user interaction.
+     * Receives (nextOpenState: boolean)
+     */
+    onInteraction: PT.func,
+
+    /** Whether to display a mask behind the popover */
+    withMask: PT.bool,
+
+    /** The position (relative to the target) at which the popover should appear. Default 'auto' */
+    position: PT.oneOf([
+        'top-left', 'top', 'top-right',
+        'right-top', 'right', 'right-bottom',
+        'bottom-right', 'bottom', 'bottom-left',
+        'left-bottom', 'left', 'left-top',
+        'auto'
+    ]),
+
+    /** Escape hatch to provide additional options to the PopperJS implementation */
+    popperOptions: PT.object
+};
+
+class LocalModel extends HoistModel {
+
+    targetRef = createObservableRef();
+    contentRef = createObservableRef();
+    @observable isOpen;
+
+    _onInteraction
+    _controlledMode = false;
+
+    get targetEl() {
+        return ReactDom.findDOMNode(this.targetRef.current);
+    }
+
+    get contentEl() {
+        return this.contentRef.current;
+    }
+
+    constructor() {
+        super();
+        makeObservable(this);
+    }
+
+    @action
+    setControlledMode(isOpen, onInteraction) {
+        this.isOpen = isOpen;
+        this._controlledMode = true;
+        if (isFunction(onInteraction)) {
+            this._onInteraction = onInteraction;
+        }
+    }
+
+    @action
+    setIsOpen(isOpen) {
+        if (this._controlledMode) {
+            this._onInteraction?.(isOpen);
```

Comment:
this is cool that this use on line 157 works!  I don't think I have ever seen this style

Reply:
Yes, I like it! Why not, right?